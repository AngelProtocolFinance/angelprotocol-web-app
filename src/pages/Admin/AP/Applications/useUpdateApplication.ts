import { EndowmentUpdateValues } from "pages/Admin/types";
import {
  EndowmentStatusNum,
  StatusChangePayload,
} from "types/server/contracts";
import { useAdminResources } from "pages/Admin/AdminGuard";
import { aws } from "services/aws/aws";
import { adminTags, awsTags } from "services/aws/tags";
import { useModalContext } from "contexts/ModalContext";
import { useGetWallet } from "contexts/WalletContext/WalletContext";
import TransactionPrompt from "components/Transactor/TransactionPrompt";
import { useSetter } from "store/accessors";
import { sendEndowmentReviewTx } from "slices/transaction/transactors";
import CW3 from "contracts/CW3";
import Registrar from "contracts/Registrar";
import cleanObject from "helpers/cleanObject";

export default function useUpdateApplicationStatus() {
  const { cw3 } = useAdminResources();
  const dispatch = useSetter();
  const { wallet } = useGetWallet();
  const { showModal } = useModalContext();

  function updateStatus(data: EndowmentUpdateValues & { PK: string }) {
    const statusChangePayload: StatusChangePayload = {
      beneficiary: data.beneficiary,
      status: +data.status as EndowmentStatusNum,
      endowment_addr: data.endowmentAddr,
    };

    const registrarContract = new Registrar(wallet);
    const embeddedMsg =
      registrarContract.createEmbeddedChangeEndowmentStatusMsg(
        cleanObject(statusChangePayload)
      );

    const contract = new CW3(wallet, cw3);
    const proposalMsg = contract.createProposalMsg(
      data.title,
      data.description,
      [embeddedMsg]
    );

    dispatch(
      sendEndowmentReviewTx({
        wallet,
        msgs: [proposalMsg],
        applicationId: data.PK,
        tagPayloads: [
          aws.util.invalidateTags([
            { type: awsTags.admin, id: adminTags.applications },
          ]),
        ],
      })
    );
    showModal(TransactionPrompt, {});
  }

  return { updateStatus };
}
