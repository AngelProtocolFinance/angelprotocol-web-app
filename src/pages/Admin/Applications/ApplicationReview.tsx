import Icon from "components/Icons/Icons";
import { CharityApplication } from "./types";

export default function ApplicationReview({
  application: ap,
}: {
  application: CharityApplication;
}) {
  return (
    <div className="bg-white-grey grid gap-2 p-4 rounded-md w-full max-w-lg max-h-75vh overflow-y-auto">
      <h1 className="font-heading font-bold text-angel-grey uppercase">
        Review Application
      </h1>
      <p>
        <span className="text-angel-grey text-sm uppercase font-bold mb-1">
          Application ID:
        </span>
        <span className="font-normal text-sm text-angel-grey ml-2">
          {ap.PK}
        </span>
      </p>
      <Field label="Name" text={ap.CharityName} />
      <Field label="Contact Email" text={ap.CharityName_ContactEmail} />
      <Field
        label="Registration Date"
        text={new Date(ap.RegistrationDate).toDateString()}
      />
      <Field label="Wallet Address" text={ap.TerraWallet} />
      <Proof
        label="Endowment Agreement"
        verified={ap.EndowmentAgreementVerified}
        link={ap.EndowmentAgreement}
      />
      <Proof
        label="Proof of Employment"
        verified={ap.ProofOfEmploymentVerified}
        link={ap.ProofOfEmployment}
      />
      <Proof
        label="Proof of Identity"
        verified={ap.ProofOfIdentityVerified}
        link={ap.ProofOfIdentity}
      />
      <div className="flex justify-center gap-4">
        <ActionButton title="Reject" actionColor="bg-failed-red" />
        <ActionButton title="Accept" actionColor="bg-bright-green" />
      </div>
    </div>
  );
}

function Field(props: { label: string; text: string }) {
  return (
    <div className="grid mb-3">
      <span className="ml-0.5 text-angel-grey text-md uppercase font-bold mb-1">
        {props.label}
      </span>
      <p
        className="p-3 text-angel-grey rounded-md bg-light-grey
         focus:outline-none"
      >
        {props.text}
      </p>
    </div>
  );
}

function Proof(props: { label: string; link: string; verified: boolean }) {
  return (
    <div className="grid mb-3">
      <span className="ml-0.5 text-angel-grey text-md uppercase font-bold mb-1">
        {props.label}
      </span>
      <a
        href={props.link}
        target="_blank"
        rel="noopener noreferrer"
        className={`focus:outline-none flex justify-between items-center gap-3 p-3 text-angel-grey rounded-md bg-light-grey overflow-hidden`}
      >
        <span className="w-3/4 truncate">{props.link}</span>
        {props.verified ? (
          <Icon type="CheckCircle" className="text-bright-green" />
        ) : (
          <Icon type="Link" />
        )}
      </a>
    </div>
  );
}

interface ActionButtonProps extends React.HTMLProps<HTMLButtonElement> {
  actionColor: string;
}

function ActionButton(props: ActionButtonProps) {
  return (
    <button
      className={`w-full tracking-wider ${props.actionColor} disabled:bg-grey-accent p-2 rounded-md mt-2 uppercase text-md text-white font-bold`}
      type="submit"
    >
      {props.title}
    </button>
  );
}
