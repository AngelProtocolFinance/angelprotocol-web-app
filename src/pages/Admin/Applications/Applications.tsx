import { createContext, useContext, useState } from "react";
import { useGetCharityApplicationsQuery } from "services/aws/registration";
import ApplicationsTable from "./ApplicationsTable";
import StatusSelector from "./StatusSelector";
import { ApplicationStatusOptions } from "./types";

export default function Applications() {
  const [applicationStatus, setApplicationStatus] =
    useState<ApplicationStatusOptions>("all");
  const {
    data = [],
    isLoading,
    isError,
  } = useGetCharityApplicationsQuery(applicationStatus);

  function handleStatusChange(ev: React.ChangeEvent<HTMLSelectElement>) {
    setApplicationStatus(ev.target.value as ApplicationStatusOptions);
  }

  return (
    <div className="px-3 pb-3 grid grid-rows-a1 bg-white/10 shadow-inner rounded-md">
      {(data.length > 0 && (
        <div className="scroll-hidden p-3 overflow-auto ">
          <div className="flex flex-row justify-between mb-10">
            <h1 className="text-2xl text-white font-semibold">
              Charity Applications
            </h1>
            <getContext.Provider value={{ activeStatus: applicationStatus }}>
              <setContext.Provider value={{ handleStatusChange }}>
                <StatusSelector />
              </setContext.Provider>
            </getContext.Provider>
          </div>
          <ApplicationsTable applications={data} isError={isError} />
        </div>
      )) || (
        <p className="font-mono text-white place-self-center mt-20">
          <span className="capitalize">
            {isLoading ? "Loading Applications..." : "No applications found"}
          </span>
        </p>
      )}
    </div>
  );
}

interface State {
  activeStatus: ApplicationStatusOptions;
}
interface Setters {
  handleStatusChange: (ev: React.ChangeEvent<HTMLSelectElement>) => void;
}

const initialState: State = {
  activeStatus: "all",
};
const getContext = createContext<State>(initialState);
const setContext = createContext<Setters>({
  handleStatusChange: () => {},
});
//only use this hook inside PhantomProvider
export const useGetApplicationsState = () => useContext(getContext);
export const useSetApplicationsState = () => useContext(setContext);

export const statusColorClasses: Record<string, string> = {
  Inactive: "bg-grey-accent",
  Active: "bg-bright-green",
  Approved: "bg-bright-green",
  "Under Review": "bg-orange",
  "Not Complete": "bg-failed-red",
};
