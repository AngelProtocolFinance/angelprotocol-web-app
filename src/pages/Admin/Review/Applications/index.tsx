import { useEndowmentApplicationsQuery } from "services/aws/registration";
import { QueryLoader } from "components/admin";
import { useGetter } from "store/accessors";
import ApplicationsTable from "./Table";

export default function Applications() {
  const { activeStatus } = useGetter((state) => state.admin.applications);
  const queryState = useEndowmentApplicationsQuery(activeStatus);

  return (
    <QueryLoader
      queryState={queryState}
      messages={{
        loading: "Loading applications...",
        error: "Failed to get applications",
      }}
    >
      {(applications) => <ApplicationsTable applications={applications} />}
    </QueryLoader>
  );
}
