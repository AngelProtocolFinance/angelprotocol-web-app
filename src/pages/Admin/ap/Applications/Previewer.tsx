import { Dialog } from "@headlessui/react";
import { CharityApplication } from "types/server/aws";
import { useModalContext } from "contexts/ModalContext";
import Icon from "components/Icon";
import PreviewForm from "./PreviewForm";

export default function Reviewer(props: { application: CharityApplication }) {
  const { closeModal } = useModalContext();

  return (
    <Dialog.Panel
      className={
        "w-full max-w-md bg-white-grey rounded-md overflow-hidden pt-4 fixed-center z-20"
      }
    >
      <button
        onClick={closeModal}
        className="absolute right-2 top-2 text-angel-grey hover:text-black"
      >
        <Icon type="Close" size={25} />
      </button>
      <PreviewForm application={props.application} />
    </Dialog.Panel>
  );
}
