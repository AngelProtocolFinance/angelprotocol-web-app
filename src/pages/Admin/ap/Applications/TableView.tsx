import { SortDirection, SortKey } from "pages/Admin/types";
import { CharityApplication } from "types/server/aws";
import { useModalContext } from "contexts/ModalContext";
import Icon from "components/Icon";
import TableSection, { Cells } from "components/TableSection";
import { statusColorClasses } from ".";
import Reviewer from "./Previewer";
import useSortedApplications from "./useSortApplications";

export default function TableView(props: {
  applications: CharityApplication[];
}) {
  const { showModal } = useModalContext();

  const openReviewer = (ap: CharityApplication) => {
    showModal(Reviewer, { application: ap });
  };

  const { sortedApplications, handleHeaderClick, sortDirection, sortKey } =
    useSortedApplications(props.applications);

  return (
    <table className="w-full text-white/80 overflow-hidden hidden sm:table">
      <TableSection type="thead" rowClass="flex flex-row block">
        <Cells
          type="th"
          cellClass="px-2 last:pr-0 flex flex-1 text-left uppercase bg-gray-200/20 py-2"
        >
          <>
            {headers.map((header) => (
              <HeaderButton
                key={header.key}
                onClick={handleHeaderClick(header.key)}
                activeSortKey={sortKey}
                sortKey={header.key}
                sortDirection={sortDirection}
              >
                {header.name}
              </HeaderButton>
            ))}
            <button className="w-full"></button>
          </>
        </Cells>
      </TableSection>
      <TableSection
        type="tbody"
        rowClass="hover:bg-angel-blue hover:bg-angel-blue/10 mb-6 sm:mb-0 flex flex-row flex-no-wrap"
      >
        {sortedApplications.map((ap) => (
          <Cells
            type="td"
            cellClass="p-2 first:pl-0 last:pr-0 group pl-4 pt-8 sm:pt-2 pb-2 text-left relative w-full border-grey-accent/30 border-t border-l border-b border-r sm:flex-1"
            key={ap.PK}
          >
            <p className="pl-1 capitalize w-42 sm:w-32">{ap.CharityName}</p>
            <p className="text-base font-bold w-42 sm:w-32 truncate">
              {ap.CharityName_ContactEmail}
            </p>
            <p className="text-base w-42 sm:w-32">
              {new Date(ap.RegistrationDate).toDateString()}
            </p>
            <p className="flex">
              <span
                className={`px-3 py-1.5 text-white-grey text-sm rounded-md ${
                  statusColorClasses[ap.RegistrationStatus]
                }`}
              >
                {ap.RegistrationStatus}
              </span>
            </p>
            <p className="font-mono flex justify-between items-center w-42 sm:w-32">
              <span className="text-base truncate w-22">
                {ap.JunoWallet || "-"}
              </span>
            </p>
            <div className="w-42 sm:w-32">
              <button
                onClick={() => openReviewer(ap)}
                className="px-3 pt-1.5 pb-1 text-white-grey bg-angel-blue hover:bg-bright-blue font-heading text-sm uppercase text-center rounded-md"
              >
                Review
              </button>
            </div>
          </Cells>
        ))}
      </TableSection>
    </table>
  );
}

const headers: { key: SortKey; name: string }[] = [
  { key: "CharityName", name: "name" },
  { key: "CharityName_ContactEmail", name: "email" },
  { key: "RegistrationDate", name: "date" },
  { key: "RegistrationStatus", name: "status" },
  { key: "JunoWallet", name: "wallet address" },
];

function HeaderButton(
  props: React.ButtonHTMLAttributes<HTMLButtonElement> & {
    sortDirection: SortDirection;
    sortKey: SortKey;
    activeSortKey: SortKey;
  }
) {
  const { activeSortKey, sortKey, sortDirection, children, ...restProps } =
    props;
  return (
    <button
      {...restProps}
      className="w-full flex items-center justify-start gap-1 uppercase font-heading font-semibold text-sm text-white/100"
    >
      <span>{children}</span>
      {activeSortKey === sortKey &&
        (sortDirection === "asc" ? <Icon type="Up" /> : <Icon type="Down" />)}
    </button>
  );
}
