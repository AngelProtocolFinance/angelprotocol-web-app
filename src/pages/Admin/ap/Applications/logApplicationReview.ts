import { createAuthToken } from "helpers";
import { LogApplicationUpdateError } from "errors/errors";
import { APIs } from "constants/urls";

interface Payload {
  PK: string;
  poll_id: string;
  chain_id: string;
}

const logApplicationReview = async (payload: Payload): Promise<void> => {
  const generatedToken = createAuthToken("charity-owner");
  const response = await fetch(`${APIs.aws}/registration?uuid=${payload.PK}`, {
    method: "PUT",
    headers: { authorization: generatedToken },
    body: JSON.stringify({
      chain_id: payload.chain_id,
      poll_id: +payload.poll_id,
    }),
  });

  //success = 2xx
  if (response.status < 200 || response.status > 299) {
    throw new LogApplicationUpdateError(payload.chain_id, payload.poll_id);
  }
};

export default logApplicationReview;
