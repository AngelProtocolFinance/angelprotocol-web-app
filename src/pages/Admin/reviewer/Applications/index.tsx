import { useState } from "react";
import { ApplicationStatusOptions, RegistrationStatus } from "types/server/aws";
import { useCharityApplicationsQuery } from "services/aws/registration";
import StatusSelector from "./StatusSelector";
import ApplicationsTable from "./Table";

export default function Applications() {
  const [applicationStatus, setApplicationStatus] =
    useState<ApplicationStatusOptions>("under-review");
  const { data = [], isLoading } =
    useCharityApplicationsQuery(applicationStatus);

  function handleStatusChange(ev: React.ChangeEvent<HTMLSelectElement>) {
    setApplicationStatus(ev.target.value as ApplicationStatusOptions);
  }

  return (
    <div className="">
      {(data.length > 0 && (
        <div className="">
          <StatusSelector
            activeStatus={applicationStatus}
            onStatusChange={handleStatusChange}
          />
          <ApplicationsTable applications={data} />
        </div>
      )) || (
        <p className="font-mono text-white place-self-center mt-20">
          <span className="capitalize">
            {isLoading ? "Loading Applications..." : "No applications found"}
          </span>
        </p>
      )}
    </div>
  );
}

export const statusColorClasses: { [key in RegistrationStatus]: string } = {
  Inactive: "bg-grey-accent",
  "Under Review": "bg-orange",
  Approved: "bg-bright-green",
  Active: "bg-bright-green",
};
