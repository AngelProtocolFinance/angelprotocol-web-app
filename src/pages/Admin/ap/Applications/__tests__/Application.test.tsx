import { StaticWalletProvider } from "@terra-money/wallet-provider";
import { NetworkInfo } from "@terra-money/wallet-provider";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { CharityApplication } from "types/server/aws";
import ModalContext from "contexts/ModalContext";
import WalletContext from "contexts/WalletContext/WalletContext";
import { store } from "store/store";
import Applications from "../Applications";

const testnet: NetworkInfo = {
  name: "testnet",
  chainID: "pisco-1",
  lcd: "https://pisco-lcd.terra.dev",
  walletconnectID: 0,
};

const mockUseGetCharityApplicationsQuery = jest.fn();
jest.mock("services/aws/registration", () => ({
  __esModule: true,
  useGetCharityApplicationsQuery: () => mockUseGetCharityApplicationsQuery(),
}));

const TestApp = (props: { routes?: string[]; initialRouteIndex?: number }) => (
  <MemoryRouter
    initialEntries={props.routes}
    initialIndex={props.initialRouteIndex}
  >
    <Provider store={store}>
      <StaticWalletProvider defaultNetwork={testnet}>
        <WalletContext>
          <ModalContext backdropClasses="z-10 fixed inset-0 bg-black/50">
            <Applications />
          </ModalContext>
        </WalletContext>
      </StaticWalletProvider>
    </Provider>
  </MemoryRouter>
);

describe("Charity applications review", () => {
  test("Renders Initial state", () => {
    mockUseGetCharityApplicationsQuery.mockReturnValue(initialState);
    render(<TestApp />);
    expect(mockUseGetCharityApplicationsQuery).toHaveBeenCalledTimes(1);
    expect(screen.getByText(/No applications found/i)).toBeInTheDocument();
  });
  test("Renders loading state", () => {
    mockUseGetCharityApplicationsQuery.mockReturnValue(loadingState);
    render(<TestApp />);
    expect(mockUseGetCharityApplicationsQuery).toHaveBeenCalledTimes(1);
    expect(screen.getByText(/Loading Applications/i)).toBeInTheDocument();
  });
  test("Application reviews rendered", () => {
    mockUseGetCharityApplicationsQuery.mockReturnValue(loadedState);
    render(<Applications />);
    expect(mockUseGetCharityApplicationsQuery).toHaveBeenCalledTimes(1);
    expect(
      screen.getByRole("heading", { name: /Charity applications/i })
    ).toBeInTheDocument();
    // assert that both application names are rendered in table cells
    expect(
      screen.getByRole("cell", {
        name: /123_company_chinadev20@outlook\.com/i,
      })
    ).toBeInTheDocument();
    expect(
      screen.getByRole("cell", {
        name: /all4good_ms@mail\.com/i,
      })
    ).toBeInTheDocument();
    // assert table has only 3 rows (heading + 2 charity applications)
    const rows = screen.getAllByRole("row");
    expect(rows.length).toBe(3);
  });

  test("Application preview modal renders 2nd application", () => {
    mockUseGetCharityApplicationsQuery.mockReturnValue(loadedState);
    render(<TestApp />);
    expect(mockUseGetCharityApplicationsQuery).toHaveBeenCalledTimes(1);
    expect(
      screen.getByRole("heading", { name: /Charity applications/i })
    ).toBeInTheDocument();
    const reviewButton = screen.getAllByRole("button", { name: /review/i })[1];
    userEvent.click(reviewButton);
    expect(
      screen.getByRole("heading", { name: /review application/i })
    ).toBeInTheDocument();
    expect(screen.getByTestId("preview-form")).toMatchSnapshot(`"123_Company"`);
  });
});

const initialState = {
  data: [],
  isLoading: false,
  isError: true,
};

const loadingState = {
  data: [],
  isLoading: true,
  isError: false,
};

const mockReviewsData: CharityApplication[] = [
  {
    ProofOfEmploymentVerified: false,
    ProofOfIdentityVerified: false,
    RegistrationDate: "2021-10-19T06:45:46.313Z",
    CharityName: "123_Company",
    EndowmentAgreementVerified: false,
    ProofOfEmployment:
      "https://charity-registration-documents.s3.amazonaws.com/proof-of-employment/eafe2aa3-6882-49f9-8f32-bc7f721fb3ab.png",
    CharityName_ContactEmail: "123_COMPANY_chinadev20@outlook.com",
    ProofOfIdentity:
      "https://charity-registration-documents.s3.amazonaws.com/proof-of-identity/eafe2aa3-6882-49f9-8f32-bc7f721fb3ab.png",
    JunoWallet:
      "juno1qsn67fzym4hak4aly07wvcjxyzcld0n4s726r2fs9km2tlahlc5qg2drvn",
    SK: "Registration",
    RegistrationStatus: "Under Review",
    PK: "eafe2aa3-6882-49f9-8f32-bc7f721fb3ab",
    EndowmentAgreement:
      "https://charity-registration-documents.s3.amazonaws.com/endowment-agreement/eafe2aa3-6882-49f9-8f32-bc7f721fb3ab.png",
  },
  {
    ProofOfEmploymentVerified: false,
    ProofOfIdentityVerified: false,
    RegistrationDate: "2021-10-19T06:49:46.313Z",
    CharityName: "All4Good",
    EndowmentAgreementVerified: false,
    ProofOfEmployment:
      "https://charity-registration-documents.s3.amazonaws.com/proof-of-employment/06c08277-bc60-4df8-b0a3-3d00e7e2f1be.png",
    CharityName_ContactEmail: "ALL4GOOD_ms@mail.com",
    ProofOfIdentity:
      "https://charity-registration-documents.s3.amazonaws.com/proof-of-identity/06c08277-bc60-4df8-b0a3-3d00e7e2f1be.png",
    JunoWallet:
      "juno1qsn67fzym4hak4aly07wvcjxyzcld0n4s726r2fs9km2tlahlc5qg243rf",
    SK: "Registration",
    RegistrationStatus: "Under Review",
    PK: "06c08277-bc60-4df8-b0a3-3d00e7e2f1be",
    EndowmentAgreement:
      "https://charity-registration-documents.s3.amazonaws.com/endowment-agreement/06c08277-bc60-4df8-b0a3-3d00e7e2f1be.png",
  },
];

const loadedState = {
  data: mockReviewsData,
  isLoading: false,
  isError: false,
};
