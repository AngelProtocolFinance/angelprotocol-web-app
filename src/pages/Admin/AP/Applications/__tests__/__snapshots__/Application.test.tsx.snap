// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Charity applications review Application preview modal renders 2nd application: "123_Company" 1`] = `
<div
  class="bg-white-grey grid gap-2 p-4 rounded-md w-full max-w-lg max-h-75vh overflow-y-auto"
  data-testid="preview-form"
>
  <h1
    class="font-heading font-bold text-angel-grey uppercase"
  >
    Review Application
  </h1>
  <p>
    <span
      class="text-angel-grey text-sm uppercase font-bold mb-1"
    >
      Application ID:
    </span>
    <span
      class="font-normal text-sm text-angel-grey ml-2"
    >
      eafe2aa3-6882-49f9-8f32-bc7f721fb3ab
    </span>
  </p>
  <div
    class="grid mb-3"
  >
    <span
      class="ml-0.5 text-angel-grey text-md uppercase font-bold mb-1"
    >
      Name
    </span>
    <p
      class="p-3 text-angel-grey rounded-md bg-light-grey focus:outline-none"
    >
      123_Company
    </p>
  </div>
  <div
    class="grid mb-3"
  >
    <span
      class="ml-0.5 text-angel-grey text-md uppercase font-bold mb-1"
    >
      Contact Email
    </span>
    <p
      class="p-3 text-angel-grey rounded-md bg-light-grey focus:outline-none"
    >
      123_COMPANY_chinadev20@outlook.com
    </p>
  </div>
  <div
    class="grid mb-3"
  >
    <span
      class="ml-0.5 text-angel-grey text-md uppercase font-bold mb-1"
    >
      Registration Date
    </span>
    <p
      class="p-3 text-angel-grey rounded-md bg-light-grey focus:outline-none"
    >
      Tue Oct 19 2021
    </p>
  </div>
  <div
    class="grid mb-3"
  >
    <span
      class="ml-0.5 text-angel-grey text-md uppercase font-bold mb-1"
    >
      Wallet Address
    </span>
    <p
      class="p-3 text-angel-grey rounded-md bg-light-grey focus:outline-none"
    >
      juno1qsn67fzym4hak4aly07wvcjxyzcld0n4s726r2fs9km2tlahlc5qg2drvn
    </p>
  </div>
  <div
    class="grid mb-3"
  >
    <span
      class="ml-0.5 text-angel-grey text-md uppercase font-bold mb-1"
    >
      Endowment Agreement
    </span>
    <a
      class="focus:outline-none flex justify-between items-center gap-3 p-3 text-angel-grey rounded-md bg-light-grey overflow-hidden"
      href="https://charity-registration-documents.s3.amazonaws.com/endowment-agreement/eafe2aa3-6882-49f9-8f32-bc7f721fb3ab.png"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        class="w-3/4 truncate"
      >
        https://charity-registration-documents.s3.amazonaws.com/endowment-agreement/eafe2aa3-6882-49f9-8f32-bc7f721fb3ab.png
      </span>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
        />
      </svg>
    </a>
  </div>
  <div
    class="grid mb-3"
  >
    <span
      class="ml-0.5 text-angel-grey text-md uppercase font-bold mb-1"
    >
      Proof of Employment
    </span>
    <a
      class="focus:outline-none flex justify-between items-center gap-3 p-3 text-angel-grey rounded-md bg-light-grey overflow-hidden"
      href="https://charity-registration-documents.s3.amazonaws.com/proof-of-employment/eafe2aa3-6882-49f9-8f32-bc7f721fb3ab.png"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        class="w-3/4 truncate"
      >
        https://charity-registration-documents.s3.amazonaws.com/proof-of-employment/eafe2aa3-6882-49f9-8f32-bc7f721fb3ab.png
      </span>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
        />
      </svg>
    </a>
  </div>
  <div
    class="grid mb-3"
  >
    <span
      class="ml-0.5 text-angel-grey text-md uppercase font-bold mb-1"
    >
      Proof of Identity
    </span>
    <a
      class="focus:outline-none flex justify-between items-center gap-3 p-3 text-angel-grey rounded-md bg-light-grey overflow-hidden"
      href="https://charity-registration-documents.s3.amazonaws.com/proof-of-identity/eafe2aa3-6882-49f9-8f32-bc7f721fb3ab.png"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        class="w-3/4 truncate"
      >
        https://charity-registration-documents.s3.amazonaws.com/proof-of-identity/eafe2aa3-6882-49f9-8f32-bc7f721fb3ab.png
      </span>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
        />
      </svg>
    </a>
  </div>
  <div
    class="flex justify-center gap-4"
  >
    <button
      class="w-full tracking-wider bg-failed-red disabled:bg-grey-accent p-2 rounded-md mt-2 uppercase text-md text-white font-bold"
      title="Reject"
      type="submit"
    >
      Reject
    </button>
    <button
      class="w-full tracking-wider bg-bright-green disabled:bg-grey-accent p-2 rounded-md mt-2 uppercase text-md text-white font-bold"
      title="Accept"
      type="submit"
    >
      Accept
    </button>
  </div>
</div>
`;
